File: zendesk_unified/zendesk_unified/android/src/main/kotlin/com/mrowl/zendesk_unified/ZendeskUnifiedPlugin.kt
package com.mrowl.zendesk_unified

import androidx.annotation.NonNull

import android.app.Activity
import io.flutter.embedding.engine.plugins.FlutterPlugin
import io.flutter.embedding.engine.plugins.FlutterPlugin.FlutterPluginBinding
import io.flutter.embedding.engine.plugins.activity.ActivityAware
import io.flutter.embedding.engine.plugins.activity.ActivityPluginBinding
import io.flutter.plugin.common.MethodCall
import io.flutter.plugin.common.MethodChannel
import io.flutter.plugin.common.MethodChannel.MethodCallHandler
import io.flutter.plugin.common.MethodChannel.Result

/** ZendeskUnifiedPlugin */
class ZendeskUnifiedPlugin: FlutterPlugin, MethodCallHandler, ActivityAware {
  /// The MethodChannel that will the communication between Flutter and native Android
  ///
  /// This local reference serves to register the plugin with the Flutter Engine and unregister it
  /// when the Flutter Engine is detached from the Activity
  private lateinit var channel : MethodChannel

  var activity: Activity? = null

  override fun onAttachedToEngine(flutterPluginBinding: FlutterPlugin.FlutterPluginBinding) {
    channel = MethodChannel(flutterPluginBinding.binaryMessenger, "plugins.com.mrowl/zendesk_unified")
    channel.setMethodCallHandler(this)
  }

  override fun onMethodCall(call: MethodCall, result: Result) {
//    val args = call.arguments as Map<String, Any>
    val zendeskApi = ZendeskApi(this,channel)
//    if (call.method == "getPlatformVersion") {
//      result.success("Android ${android.os.Build.VERSION.RELEASE}")
//    } else {
//      result.notImplemented()
//    }
    if(call.method == "initialize") {
      try {
        result.success(zendeskApi.initialize())
      } catch (e: Exception) {
        // Print the exception to the console
        e.printStackTrace()
        result.error("Error", e.message, null)
      }
      result.success(zendeskApi.initialize())
    }
    if(call.method == "setAnonymousIdentity") {
      result.success(zendeskApi.setAnonymousIdentity())
    }
    if(call.method == "showHelpCenter") {
      result.success(zendeskApi.showHelpCenter())
    }
    else {
      result.notImplemented()
      }
  }




  

  override fun onDetachedFromEngine(binding: FlutterPlugin.FlutterPluginBinding) {
    channel.setMethodCallHandler(null)
  }

    override fun onAttachedToActivity(binding: ActivityPluginBinding) {
        activity = binding.activity
    }

    override fun onDetachedFromActivity() {
        activity = null
    }

    override fun onReattachedToActivityForConfigChanges(binding: ActivityPluginBinding) {
        activity = binding.activity
    }

    override fun onDetachedFromActivityForConfigChanges() {
        activity = null
    }
}

File: zendesk_unified/zendesk_unified/ios/Classes/ZendeskUnifiedPlugin.swift
import Flutter
import UIKit

public class ZendeskUnifiedPlugin: NSObject, FlutterPlugin {
    private var channel: FlutterMethodChannel
    
    // Init
    init(channel: FlutterMethodChannel) {
        self.channel = channel
    }
    
    
  public static func register(with registrar: FlutterPluginRegistrar) {
    // For platform view
    let factory = ZendeskNativeViewFactory(messenger: registrar.messenger())
    registrar.register(factory, withId: "plugins.com.mrowl/zendesk_unified_view")
    let channel = FlutterMethodChannel(name: "plugins.com.mrowl/zendesk_unified", binaryMessenger: registrar.messenger())
    let instance = ZendeskUnifiedPlugin(channel: channel)
    registrar.addMethodCallDelegate(instance, channel: channel)
    registrar.addApplicationDelegate(instance)
  }

  public func handle(_ call: FlutterMethodCall, result: @escaping FlutterResult) {
      let zendeskApi = ZendeskApi(flutterPlugin: self, channel:channel)
    switch call.method {
    case "initialize":
        result(zendeskApi.initialize())
    case "setIdentity":
        let args = call.arguments as! [String: Any]
        let id = args["id"] as! String
        
        result(zendeskApi.setIdentity(email: id))
    case "showHelpCenter":
        
        result(zendeskApi.showHelpCenter())
    case "closeHelpCenter":
        result(zendeskApi.closeHelpCenter())
    default:
      result(FlutterMethodNotImplemented)
    }
  }
}
